hoot-core-test/src/test/cpp/hoot/core/ops/UnconnectedWaySnapperTest.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2019 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+// Hoot
+#include <hoot/core/elements/OsmMap.h>
+#include <hoot/core/TestUtils.h>
+#include <hoot/core/ops/UnconnectedWaySnapper.h>
+#include <hoot/core/util/Log.h>
+#include <hoot/core/util/MapProjector.h>
+#include <hoot/core/io/OsmXmlReader.h>
+#include <hoot/core/io/OsmXmlWriter.h>
+#include <hoot/core/criterion/StatusCriterion.h>
+#include <hoot/core/visitors/RemoveElementsVisitor.h>
+
+// CPP Unit
+#include <cppunit/extensions/HelperMacros.h>
+#include <cppunit/extensions/TestFactoryRegistry.h>
+#include <cppunit/TestAssert.h>
+#include <cppunit/TestFixture.h>
+
+namespace hoot
+{
+
+class UnconnectedWaySnapperTest : public HootTestFixture
+{
+  CPPUNIT_TEST_SUITE(UnconnectedWaySnapperTest);
+  CPPUNIT_TEST(runSnapTest);
+  CPPUNIT_TEST(runConfigOptionsValidationTest);
+  CPPUNIT_TEST_SUITE_END();
+
+public:
+
+  const QString inputPath = "test-files/ops/UnconnectedWaySnapper/";
+  const QString outputPath = "test-output/ops/UnconnectedWaySnapper/";
+
+  UnconnectedWaySnapperTest()
+  {
+    setResetType(ResetBasic);
+    TestUtils::mkpath(outputPath);
+  }
+
```

Comment:
A little more thorough testing *could* be done here but am holding off until after seeing how well it works for Tradecraft.